apps/meteor/server/startup/migrations/v305.ts

```diff
@@ -0,0 +1,22 @@
+import { Settings } from '@rocket.chat/models';
+
+import { addMigration } from '../../lib/migrations';
+
+const maxAgeSettings = ['RetentionPolicy_MaxAge_Channels', 'RetentionPolicy_MaxAge_Groups', 'RetentionPolicy_MaxAge_DMs'];
+
+const convertDaysToMs = (days: number) => days * 24 * 60 * 60 * 1000;
+addMigration({
+	version: 305,
+	name: 'Convert retention policy max age from days to milliseconds',
+	async up() {
+		// if the package value isn't 30 the setting has already been updated
+		await Settings.find({ _id: { $in: maxAgeSettings }, packageValue: 30 }, { projection: { _id: 1, value: 1 } })
```

Comment:
I think it would be better to retrieve all the rows and store them in a variable, then use Promise.all to update all the values in parallel

Reply:
This was being done just like described. @KevLehman made some changes in commit [ee32132](https://github.com/RocketChat/Rocket.Chat/pull/32425/commits/ee321325787018fbb3efea786f516e423bd32057) that removed the Promise.all

Reply:
I changed this to be this way because at max, this thing will process 3 entries. So doing them in parallel or a bulk write doesn't give any extra perf gain